climada/engine/impact.py

```diff
@@ -1389,6 +1389,184 @@ def _selected_exposures_idx(self, coord_exp):
             LOGGER.warning("No exposure coordinates match the selection.")
         return sel_exp
 
+    @classmethod
+    def concat(cls, imp_list):
+        """Concatenate impact objects of the same hazard type, frequency,
+        exposure, and coordinate system. This function is useful if, e.g. different impact functions
+        have to be applied for different seasons (e.g. for agricultural impacts).
+
+        NOTES
+        -----
+            - Concatenation of impacts with different exposure (e.g. different countries)
+                could also be implemented here in the future.
+
+        Parameters
+        ----------
+        imp_list : list of climada.engine.Impact objects
+                list of Impact object to concatenate
+
+        Returns
+        --------
+        impact: climada.engine.Impact
+               New impact object which is a concatenation of all impacts
+        """
+
+        # checks
+        def check_haz_types(imp_list):
+            haz_types = [imp.tag['haz'].haz_type for imp in imp_list if \
+                         imp.tag['haz'].haz_type != '']
+            if len(haz_types) < len(imp_list):
+                raise ValueError(
+                    "Not all impacts have a hazard type."
+                    "Please specify the same hazard type for each impact."
+                    "The impacts are incompatible and cannot be concatenated.")
```

Comment:
I do not really like the sentence you added to all errors `The impacts are incompatible and cannot be concatenated.` What is that supposed to mean?

Reply:
I second this. The error traceback shows that it occurs within the `concat` method, so there is no need to mention that it did not work as expected 😉